src/Expander.hs

```diff
@@ -42,13 +43,13 @@ import Control.Applicative
 import Control.Lens hiding (List, children)
 import Control.Monad.Except
 import Control.Monad.Reader
-import Control.Monad.Writer
+import Control.Monad.State.Strict
```

Comment:
Why not Control.Monad.Writer.CPS? It keeps the Writer API, making it clear that computations cannot read and be affected by the state, while using a State-based implementation under the hood.

Reply:
Ah I was not aware of `Writer.CPS`!

Reply:
implemented in 40d4fccd20e4e43123cb415b64c4b251239022c3